src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -342,4 +399,14 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
     }
   }
 
+  private final class SimplePartitionerImpl implements SimplePartitioner {
```

Comment:
I'm not sure about this either.  That's fine we can remove it, but we should document the behavior when null keys are used.